### PR TITLE
M4 - Import wallet file as mina_signer::Keypair

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**

This PR adds a crate that can load keypairs generated by [mina-generate-keypair](https://docs.minaprotocol.com/en/using-mina/keypair) tool, into [mina_signer::Keypair](https://o1-labs.github.io/proof-systems/rustdoc/mina_signer/keypair/struct.Keypair.html)

Changes introduced in this pull request:
- secret box loader
- unit tests
- E2E test that loads a real wallet file and validate its public key
- wasm tests

**Reference issue to close (if applicable)**

Closes https://github.com/ChainSafe/mina-rs/issues/100

**Other information and links**

### CODE REVIEW

File: secrets/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+[dependencies]
+proof-systems = {path = "../proof-systems-shim"}
+
+argon2 = "0.4"
+bs58 = {version = "0.4", features = ["check"]}
+hex = "0.4"
+serde = {version = "1", features = ["derive"]}
+serde_json = "1"
+thiserror = "1"
+xsalsa20poly1305 = "0.8"
```
Comment (willemolding) [inline secrets/Cargo.toml:24]: Does adding these add much to our build times? I know sometimes these crypto libs can be a big burden

Comment (hanabi1224) [reply secrets/Cargo.toml:24]: These 2 new crypto libs are both implemented in pure rust, and the CI task does not show notable increase in build time.  Currently the slowest dependency is C code from rocksdb-sys which adds several minutes to CI build time

Comment (hanabi1224) [reply secrets/Cargo.toml:24]: Here's a per-crate build report of `secrets` generated with `cargo-bloat`

  Time Crate
13.37s syn
11.48s bitvec
10.26s ark_ff
 8.85s rayon
 8.62s num_bigint
 7.55s rayon_core
 7.36s num_bigint
 6.88s serde_derive
 5.85s rustversion
 5.38s libc
 4.72s crossbeam_utils
 4.62s num_traits
 4.58s num_traits
 4.38s serde
 4.32s proc_macro2
 4.10s generic_array
 4.09s rand
 3.91s crossbeam_epoch
 3.77s sha2
 3.77s sha2
 3.65s darling_core
 3.62s serde_json
 3.42s num_integer
 3.32s blake2
 3.20s paste
 3.20s ahash
 3.18s radium
 3.15s pest
 3.12s funty
 3.11s derivative
 3.02s num_integer
 **2.97s argon2**
 2.82s crossbeam_channel
 2.77s build_script_main
 2.74s thiserror_impl
 2.67s password_hash
 2.47s typenum
 2.19s poly1305
 2.17s rand_chacha
 2.11s memoffset
 1.98s semver_parser
 1.93s ark_ff_macros
 1.91s serde_with
 1.83s hashbrown
 1.72s base64ct
 1.72s ark_ff_asm
 1.66s bs58
 1.63s zeroize_derive
 1.61s ppv_lite86
 1.58s ark_serialize_derive
 1.58s num_cpus
 1.48s ucd_trie
 1.45s autocfg
 1.43s ryu
 1.40s version_check
 1.30s oracle
 1.27s quote
 1.27s strsim
 1.25s rand_core
 1.19s hex
 1.16s digest
 1.15s ark_poly
 1.13s rustc_version
 1.07s getrandom
 1.06s semver
 1.00s mina_secrets
 0.99s mina_signer
 0.97s wyz
 0.97s ark_ec
 0.93s serde_with_macros
 0.91s synstructure
 0.87s aead
 0.78s either
 0.77s digest
 0.75s mina_hasher
 0.75s cipher
 0.73s crossbeam_deque
 0.69s once_cell
 0.67s subtle
 0.66s unicode_xid
 0.65s ark_serialize
 0.62s darling_macro
 0.60s block_buffer
 0.60s lazy_static
 0.57s salsa20
 0.57s block_buffer
 0.57s itoa
 0.54s ident_case
 0.53s crypto_common
 0.52s tap
 0.51s universal_hash
 0.50s scopeguard
 0.49s cpufeatures
 0.47s opaque_debug
 0.46s bcs
 0.45s fnv
 **0.45s xsalsa20poly1305**
 0.43s zeroize
 0.40s cfg_if
 0.36s o1_utils
 0.32s ark_std
 0.29s mina_curves
 0.21s proof_systems
 0.20s thiserror
 0.12s darling

File: secrets/src/secret_box/types_impls.rs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        json.try_into()
+    }
+}
+
+impl TryFrom<SecretBox> for SecretBoxJson {
+    type Error = Error;
+    fn try_from(sb: SecretBox) -> Result<SecretBoxJson, Self::Error> {
+        let pwsalt = {
+            // 16 is sufficient here, alloc array on stack instead
+            // let mut buf = vec![0; self.pwsalt.len()];
```
Comment (creativcoder) [inline secrets/src/secret_box/types_impls.rs:51]: nit: can we remove this?

**Review by sudo-shashank — APPROVED** [review]

File: secrets/src/secret_box/types_impls.rs (hunk 2 above)
Comment (hanabi1224) [reply secrets/src/secret_box/types_impls.rs]: fixed

**Review by creativcoder — APPROVED** [review]

**Pull request merged**